openmls/src/group/tests/test_proposal_validation.rs

```diff
@@ -0,0 +1,190 @@
+//! This module tests the validation of proposals as defined in
+//! https://openmls.tech/book/message_validation.html#semantic-validation-of-proposals-covered-by-a-commit
+
+use openmls_rust_crypto::OpenMlsRustCrypto;
+use openmls_traits::OpenMlsCryptoProvider;
+
+use rstest::*;
+use rstest_reuse::{self, *};
+
+use crate::{
+    config::*,
+    credentials::*,
+    group::errors::*,
+    group::*,
+    key_packages::*,
+    prelude_test::{FramingParameters, MlsPlaintext, WireFormat},
+};
+
+/// Helper function to generate a CredentialBundle
+fn generate_credential_bundle(
```

Comment:
I used those functions as well for external commit validation and I think they're generally useful functions for testing. Not necessarily something we need to do in this PR, but maybe we should keep them in `utils.rs` or some other place, where they're more accessible?

Reply:
Agreed, we should do that in a follow-up. Do you want to file an issue?

Reply:
I don't think this needs an explicit issue. I'll do it in the external commit validation PR in a second, which I would expect will be merged before this one.

Reply:
Sounds good, once that lands, I can merge and remove the helper functions here.

Reply:
The PR has landed. Once you merge `main` into this branch, you should be able to find the `generate...` functions in the `utils.rs` submodule next to this one.